proxy/src/main/java/com/wavefront/agent/api/ForceQueueEnabledProxyAPI.java

```diff
@@ -0,0 +1,74 @@
+package com.wavefront.agent.api;
+
+import java.util.List;
+import java.util.UUID;
+
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+
+/**
+ * Wrapper around WavefrontV2API that supports forced writing of tasks to the backing queue.
+ *
+ * @author Andrew Kao (akao@wavefront.com)
+ * @author vasily@wavefront.com
+ */
+public interface ForceQueueEnabledProxyAPI extends WavefrontV2API {
+
+  /**
+   * Report batched data (metrics, histograms, spans, etc) to Wavefront servers.
+   *
+   * @param proxyId       Proxy Id reporting the result.
+   * @param format        The format of the data (wavefront, histogram, trace, spanLogs)
+   * @param pushData      Push data batch (newline-delimited)
+   * @param forceToQueue  Whether to bypass posting data to the API and write to queue instead.
+   */
+  Response proxyReport(@HeaderParam("X-WF-PROXY-ID") final UUID proxyId,
```

Comment:
just a question: this is an API that is on the proxy _input_ side, yes?

Reply:
@jbau: that's the internal wrapper "API" over proxy's output API that just adds another argument to support writing directly to the on-disk queue instead of sending. 
The output API is in java-lib/src/main/java/com/wavefront/api/ProxyV2API.java - I believe this is a perfect moment to separate these methods into a separate interface so we can eventually deprecate the old ones; and while we're at it, get rid of legacy arguments that no longer make sense.